prowler/providers/gcp/gcp_provider.py

```diff
@@ -238,12 +243,52 @@ def setup_session(self, credentials_file: str, service_account: str) -> Credenti
             )
             sys.exit(1)
 
-    def __set_gcp_creds_env_var__(self, credentials_file):
-        logger.info(
-            "GCP provider: Setting GOOGLE_APPLICATION_CREDENTIALS environment variable..."
-        )
-        client_secrets_path = os.path.abspath(credentials_file)
-        os.environ["GOOGLE_APPLICATION_CREDENTIALS"] = client_secrets_path
+    @staticmethod
+    def test_connection(
+        credentials_file: str = None,
+        service_account: str = None,
+        raise_on_exception: bool = True,
+    ) -> Connection:
+        """
```

Comment:
Could you add docstrings following the guide from https://github.com/google/styleguide/blob/gh-pages/pyguide.md#38-comments-and-docstrings please? Thanks!